Functions/xASL_bids_CreateDatasetDescriptionTemplate.m

```diff
@@ -0,0 +1,105 @@
+%   License             - License (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%   Authors             - Authors (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%   Acknowledgements    - Acknowlegements (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%
+% OUTPUT:
+%   json                - JSON structure which can be saved using spm_jsonwrite (dataset_description.json template)
+%                         
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  This script creates a JSON structure which can be saved
+%               using spm_jsonwrite to get a dataset_description.json template.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:      [json] = xASL_bids_CreateDatasetDescriptionTemplate('Test Patient','1.5.0','1.2.3','Test','Test License','Author Names','Thanks');
+%               [json] = xASL_bids_CreateDatasetDescriptionTemplate();
+%               
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    % Get default BIDS configuration
+    bidsPar = xASL_bids_Config();
+
+    % Create dummy dataset_description.json
+    json = struct;
+    
+    % Required fields
+    for iCell = 1:numel(bidsPar.datasetDescription.Required)
+         json.(bidsPar.datasetDescription.Required{1,iCell}) = 'Undefined';
+    end
+    % Recommended fields
+    for iCell = 1:numel(bidsPar.datasetDescription.Recommended)
+         json.(bidsPar.datasetDescription.Recommended{1,iCell}) = 'Undefined';
+    end
+    % Optional fields
+    for iCell = 1:numel(bidsPar.datasetDescription.Required)
+         json.(bidsPar.datasetDescription.Optional{1,iCell}) = 'Undefined';
+    end
+    
+    % Parse the input parameters
+    p = inputParsing(varargin{:});
+    
+    % Optionally change the parameter values
+    json.Name = p.Results.Name;
+    json.BIDSVersion = p.Results.BIDSVersion;
+    json.HEDVersion = p.Results.HEDVersion;
+    json.DatasetType = p.Results.DatasetType;
+    json.License = p.Results.License;
+    json.Authors = p.Results.Authors;
+    json.Acknowledgements = p.Results.Acknowledgements;
+    
+end
+
+%% Parse input
+function p = inputParsing(varargin)
+
+    % Get default BIDS configuration
+    bidsPar = xASL_bids_Config();
+
+    % Initialize input parser
+    p = inputParser;
+    
+    % Define valid input variables
+    validName = @(variable) ischar(variable);
+    validBIDSVersion = @(variable) ischar(variable);
+    validHEDVersion = @(variable) ischar(variable);
+    validDatasetType = @(variable) ischar(variable);
+    validLicense = @(variable) ischar(variable);
+    validAuthors = @(variable) ischar(variable);
+    validAcknowledgements = @(variable) ischar(variable);
+    
+    % Define defaults
+    defaultName = 'Undefined';
```

Comment:
Does not make sense to define here the default values etc. Because it doesn't follow the bids_Config, and also it is done already above in the parent function. 
But my recommendation would be to get rid of this parameter parsing anyway.

Reply:
Fixed in bcfe3e6.